src/promptflow/promptflow/_utils/tool_utils.py

```diff
@@ -129,3 +130,30 @@ def get_inputs_for_prompt_template(template_str):
 def get_prompt_param_name_from_func(f):
     """Get the param name of prompt template on provider."""
     return next((k for k, annotation in f.__annotations__.items() if annotation == PromptTemplate), None)
+
+
+def get_input_type_mapping_for_prompt_template(template_str):
+    """Get all tagged input type mapping from a template string"""
+
+    # currently we only support image type
+    pattern = r'\!\[(\s*image\s*)\]\(\{\{\s*([^{}]+)\s*\}\}\)'
+    matches = re.finditer(pattern, template_str)
+
+    result_dict = {}
+
+    for match in matches:
+        input_name = match.group(2).strip()
+        if input_name != "" and input_name not in result_dict:
+            result_dict[input_name] = match.group(1).strip()
+
+    # try best to identify object types
+    try:
+        env = Environment()
+        template = env.parse(template_str)
+        for node in template.find_all(node_type=nodes.For):
```

Comment:
What does `nodes.For` mean?

Reply:
nodes.For target the variable that was wrapped with {% %} in jinja. I am trying to identify these variables and treat it as object type. Anyway, we can leave this change to a separate PR with more discussion.